source/common/http/conn_manager_utility.cc

```diff
@@ -102,8 +102,10 @@ Network::Address::InstanceConstSharedPtr ConnectionManagerUtility::mutateRequest
   // HUGE WARNING: The way we do this is not optimal but is how it worked "from the beginning" so
   //               we can't change it at this point. In the future we will likely need to add
   //               additional inference modes and make this mode legacy.
-  const bool internal_request = single_xff_address && final_remote_address != nullptr &&
-                                Network::Utility::isInternalAddress(*final_remote_address);
+  const bool internal_request =
+      (single_xff_address && final_remote_address != nullptr) &&
+      (Network::Utility::isInternalAddress(*final_remote_address) ||
```

Comment:
I think this should be reworked so that it only calls config.internalAddressConfig().isInternalAddress(), and in the case that this bit is unconfigured, we have an implementation of the interface that calls Network::Utility::isInternalAddress(). Because if/when we add different subnets to that config, we won't want to ever use the current behavior of Utility::isInternalAddress().

Reply:
Yeah thinking about it some more that makes a lot of sense. With the current approach there's no way to disable bits of the existing behavior, which I'm sure someone would find useful.